### PR TITLE
fix: sync squad MLS state across peers and stop raw JSON in UI

### PR DESCRIPTION
**Pull request by daopunk**
## Summary

Squad peers can now catch up on roster EVM and governance announces after joining, deploy governance as captain even when Default only funds gas, and see Crew/Status match the bound squad key (and on-chain Hats) instead of the WalletBar Default. MLS dashboard mirrors and DMs also stop dumping raw JSON, and hat-less members keep Governance **read** access while hats still gate writes.

**Before:** Late joiners missed older EVM/infra shares; Default-as-payer could not complete hats sponsor (captain ACL); state-sync could overwrite the shared roster row with Default so Crew showed no Captain hat while Hats Protocol was correct; join/ACL surfaces showed raw JSON.

**After:** Join/sync republishes bound roster + announces; Default→squad fund then deploy-as-captain; publish/heal keep share aligned with binding; peers and UI stay in sync without remount; errors and join DMs stay human-readable.

## Changes

- Persist `squad_member_evm_share` from #announcements; widen MLS side-effect trust to group members; FE `onMlsStructuredMessage` refresh for infra/EVM/allowlist/tokens/join.
- Passive squad state sync: auto-request on join, peer republish (EVM + announces), manual Request sync on My Dashboard → Status.
- Publish roster share prefers **bound** squad address over active Default; heal share↔binding on roster load; exclude payer-only Default from crew bootstrap.
- Deploy: captain locked to squad-assigned EVM; Default can transfer ETH to fund that key, then gov+sponsor sign as squad; roster alt-parent / binding heal for captain validation.
- DM cards for bot join request/response; unwrap `wallet_err_json`; structured-JSON safety net; optimistic mutates roll back (or publish-first) on announce failure.
- Docs: peer-sync contract in `docs/communities/DESIGN.md` / ADR; deploy smoke notes in `docs/wallet/OPERATOR_SMOKE.md`.

## Test plan

- [ ] `pnpm exec vitest run src/lib/utils/tauri-errors.test.ts src/lib/dm/resolve-dm-message-presentation.test.ts src/lib/app/mls-structured-refresh.test.ts src/lib/app/tauri-subscriptions.test.ts src/lib/squad/squad-state-sync.test.ts src/lib/squad/squad-member-evm-share.test.ts`
- [ ] Two clients: A binds EVM → B Crew shows A without remount
- [ ] New member joins → auto sync request; existing member republishes; late joiner sees roster + gov announces
- [ ] Privacy deploy: empty squad key + funded Default → transfer amount → deposit &lt; transfer → captain hat on squad key; Crew/Status show squad key (not Default)
- [ ] After bad Default overwrite: open Crew/Status → heal republishes bound address; Captain hat appears
- [ ] Join accept DM shows accepted/rejected notice, not JSON; soft-fail sponsor toast is plain language
- [ ] Hat-less member can open Governance (read); write CTAs stay disabled with reason

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

File: src-tauri/src/db.rs [hunk 1]
```diff
@@ -1174,31 +1174,41 @@ pub fn side_effect_parent_matches_chat(chat_id: &str, payload_parent_id: &str) -
...
+            }
+        }
+    }
+
     let Ok(conn) = crate::account_manager::get_db_connection(handle) else {
         return false;
     };
-    let creator: Option<String> = conn
+    let known: bool = conn
         .query_row(
```
Comment (logicminds) [inline src-tauri/src/db.rs]: P1 test comment.

**Review by logicminds — CHANGES_REQUESTED** [review]
Test review

File: src-tauri/src/db.rs [hunk 2]
```diff
@@ -1174,31 +1174,41 @@ pub fn side_effect_parent_matches_chat(chat_id: &str, payload_parent_id: &str) -
     !chat.is_empty() && !parent.is_empty() && chat == parent
 }
 
-/// True when `author_npub` is the MLS group creator for `chat_id`.
-/// Interim v1: only the group creator is an MLS admin; reject if the group is unknown.
-fn is_author_mls_admin_for_chat<R: Runtime>(
+/// True when `author_npub` is treated as an MLS member of `chat_id` for announce side effects.
+/// Prefer in-memory participants when populated; otherwise a known `mls_groups` row plus
+/// non-empty author (MLS delivery already authenticated the sender).
+fn is_author_mls_member_for_chat<R: Runtime>(
```
Comment (logicminds) [inline src-tauri/src/db.rs:1220]: **P2 — Testing.** The new `is_author_mls_member_for_chat` DB fallback path (empty in-memory participants, state lock failure, removed-member replay) has no test coverage.

Suggested fix: add Rust unit tests for this fallback and for the empty-participants case.

File: src-tauri/src/db.rs [hunk 3]
```diff
@@ -1174,31 +1174,41 @@ pub fn side_effect_parent_matches_chat(chat_id: &str, payload_parent_id: &str) -
...
+
     let Ok(conn) = crate::account_manager::get_db_connection(handle) else {
         return false;
     };
-    let creator: Option<String> = conn
+    let known: bool = conn
         .query_row(
-            "SELECT creator_pubkey FROM mls_groups WHERE group_id = ?1",
-            rusqlite::params![chat_id.trim()],
+            "SELECT EXISTS(SELECT 1 FROM mls_groups WHERE group_id = ?1 OR engine_group_id = ?1)",
```
Comment (logicminds) [inline src-tauri/src/db.rs]: **P1 — Security/Correctness.** The DB fallback only verifies that a `mls_groups` row exists; it does not check that the author is actually a member of the group. If the in-memory state is missing or the lock fails, any non-empty `author_npub` can trigger allowlist, tracked-token, treasury, and governance side effects.

Suggested fix: query `chats.participants` (or another membership source) in the fallback, or fail closed when membership cannot be verified.

File: src/lib/squad/squad-state-sync.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  squadInfraLegacyProvider,
+} from '../governance/api';
+import { currentUser } from '../../stores/auth';
+import { publishSquadMemberEvmShare } from './squad-member-evm-share';
+
+export const SQUAD_STATE_SYNC_REQUEST_TYPE = 'squad_state_sync_request';
+export const SQUAD_STATE_SYNC_REQUEST_VERSION = 1;
+
+const RESPOND_COOLDOWN_MS = 15_000;
+const respondedRequestKeys = new Set<string>();
```
Comment (logicminds) [inline src/lib/squad/squad-state-sync.ts:27]: **P3 — Maintainability.** `respondedRequestKeys` is an unbounded global `Set`; memory grows over a long-lived session.

Suggested fix: cap or prune the set after a reasonable number of entries.

File: src/lib/squad/squad-state-sync.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  return { parent_id, request_id, requester_npub, requested };
+}
+
+/** Test helper: clear in-memory debounce state. */
+export function resetSquadStateSyncRespondStateForTests(): void {
+  respondedRequestKeys.clear();
+  lastRespondAtByParent.clear();
+}
+
+function autoRequestStorageKey(announcementsGroupId: string): string {
```
Comment (logicminds) [inline src/lib/squad/squad-state-sync.ts]: **P2 — Project Standards.** `autoRequestStorageKey` uses `sessionStorage` without scoping by npub. Switching accounts in the same browser session can skip the auto-request for the second account.

Suggested fix: prefix the key with the current npub so account switches do not suppress the request.

File: src/lib/squad/squad-state-sync.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+  if (!me) return false;
+  try {
+    await syncMlsGroupsNow(gid);
+  } catch {
+    /* still attempt publish */
+  }
+  const requestId =
+    typeof crypto !== 'undefined' && typeof crypto.randomUUID === 'function'
+      ? crypto.randomUUID()
+      : `sync-${Date.now()}`;
```
Comment (logicminds) [inline src/lib/squad/squad-state-sync.ts]: **P3 — Reliability.** The `crypto.randomUUID` fallback uses `sync-${Date.now()}`, which can collide if two requests are created in the same millisecond.

Suggested fix: add a counter or random suffix to the fallback ID.

File: src/lib/squad/squad-state-sync.ts [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+  const req = parseSquadStateSyncRequest(content);
+  if (!req) return;
+  const gid = groupId.trim() || req.parent_id;
+  if (!gid) return;
+  const me = get(currentUser)?.npub?.trim();
+  if (!me) return;
+  if (req.requester_npub === me) return;
+
+  const parentId = req.parent_id || gid;
+  const respondKey = `${parentId}:${req.request_id}`;
```
Comment (logicminds) [inline src/lib/squad/squad-state-sync.ts:182]: **P3 — Security/Reliability.** `respondToSquadStateSyncRequest` does not reject a request whose `payload.parent_id` differs from the MLS `groupId`. If a malformed/crafted sync request arrives with a mismatched parent, the responder may republish state for the wrong parent.

Suggested fix: assert `req.parent_id === gid` before republishing.

File: src/lib/squad/squad-state-sync.ts [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+  if (respondedRequestKeys.has(respondKey)) return;
+  const last = lastRespondAtByParent.get(parentId) ?? 0;
+  if (Date.now() - last < RESPOND_COOLDOWN_MS) return;
+  respondedRequestKeys.add(respondKey);
+  lastRespondAtByParent.set(parentId, Date.now());
```
Comment (logicminds) [inline src/lib/squad/squad-state-sync.ts]: **P2 — Reliability.** `lastRespondAtByParent` is updated before the EVM/infra republish runs. If the publish fails, the next legitimate request is still suppressed for `15s`.

Suggested fix: move the cooldown update to after a successful republish, or track it per request.

File: src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte [hunk 9]
```diff
@@ -235,6 +301,30 @@
     };
 
     deploying = true;
+
+    if (needsFundTransfer && transferWei != null && squadCanonical && squadNetwork) {
+      progressStep = 'fund';
+      const send = await walletBuildAndSendTransaction(
+        '',
+        squadNetwork,
+        'ETH',
```
Comment (logicminds) [inline src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte:310]: **P2 — Correctness.** The Default→squad fund transfer passes the raw `transferTrimmed` string to `walletBuildAndSendTransaction`, while the rest of the validation uses the parsed `transferWei` bigint.

Suggested fix: pass the already-validated `transferWei.toString()` to avoid parse drift.

**Review by logicminds — CHANGES_REQUESTED** [review]
Code review summary — automated/manual review.

This PR adds late-joiner MLS squad state catch-up, Default→squad funding for captain deploys, and fixes raw JSON rendering in DM/MLS chat. The change is large and touches authorship gates, new sync protocol state, and treasury/governance deploy wiring.

**Verdict:** ready with fixes. The P1 membership fallback is the merge blocker; the P2 items are straightforward follow-ups.

All frontend tests (`159` files / `1371` tests) and Rust library tests (`299` passed / `1` ignored) pass. The multi-agent reviewer flow could not start in this environment, so this is a manual focused review of the highest-risk paths.

**Review by logicminds — APPROVED** [review]
